gateway/src/main/java/com/mx/path/gateway/configuration/ConfigurationBinder.java

```diff
@@ -273,7 +273,8 @@ private void validate(Object obj) {
     List<Annotations.AnnotatedField<ConfigurationField>> annotatedFields = Annotations.fieldsWithAnnotation(ConfigurationField.class, obj.getClass());
     annotatedFields.forEach(annotatedField -> {
       Object value = Fields.getFieldValue(annotatedField.getField(), obj);
-      state.withField(annotatedField.getField().getName(), () -> validateField(annotatedField, value));
+      String field = Strings.isBlank(annotatedField.getAnnotation().value()) ? annotatedField.getField().getName() : annotatedField.getAnnotation().value();
```

Comment:
A unit test for the `annotatedField.getField().getName()` code branch would be awesome too!

Reply:
Added a test. Let me know if that's what you had in mind. Also got the commit history cleaned up.